backend/spec/requests/api/v1/reset_password_spec.rb

```diff
@@ -0,0 +1,141 @@
+
+          it "returns 200" do
+            expect(response).to have_http_status(:ok)
+          end
+        end
+
+        context "wrong params" do
+          let(:params) { {} }
+
+          it "returns 200" do
+            expect(response).to have_http_status(:ok)
+          end
+        end
+      end
+    end
+
+    put "Resets password" do
+      tags "Reset Password"
+      consumes "application/json"
+      produces "application/json"
+      security [csrf: []]
+      parameter name: :params, in: :body, schema: {
+        type: :object,
+        properties: {
+          password: {type: :string},
+          reset_password_token: {type: :string}
+        },
+        required: ["email", "reset_password_token"]
+      }
+
+      response "200", :success do
+        schema type: :object, properties: {
+          data: {"$ref" => "#/components/schemas/user"}
+        }
+        let(:token) {
+          user.send(:set_reset_password_token) # this is protected method
+        }
+        let(:params) { {password: "NewPassword1234", reset_password_token: token} }
+        let("X-CSRF-TOKEN") { get_csrf_token }
+
+        run_test!
+
+        it "changes user password and signs in user" do
+          user.reload
+          expect(user.valid_password?("NewPassword1234")).to eq(true)
+          expect(user.reset_password_token).to be_nil
+          expect(user.reset_password_sent_at).to be_nil
+        end
+      end
+
+      response "422", "Invalid, expired token" do
+        let(:token) { "invalid" }
+        let(:params) { {password: "NewPassword1234", reset_password_token: token} }
+        let("X-CSRF-TOKEN") { get_csrf_token }
+
+        run_test!
+
+        it "returns correct error", generate_swagger_example: true do
+          expect(response_json["errors"][0]["title"]).to eq("Reset password token is invalid")
+        end
+
+        context "no token" do
+          let(:params) { {} }
+
+          it "returns correct error", generate_swagger_example: true do
+            expect(response_json["errors"][0]["title"]).to eq("Reset password token can't be blank")
+          end
+        end
+
+        context "expired token" do
+          let(:token) do
+            travel_to 7.hours.ago do
+              user.send(:set_reset_password_token) # this is protected method
+            end
+          end
+          let(:params) { {password: "NewPassword1234", reset_password_token: token} }
+
+          it "returns correct error", generate_swagger_example: true do
+            expect(response_json["errors"][0]["title"]).to eq("Reset password token has expired, please request a new one")
+          end
+        end
+
+        context "invalid password" do
+          let(:token) {
+            user.send(:set_reset_password_token) # this is protected method
+          }
+          let(:params) { {password: "secret", reset_password_token: token} }
+
+          it "returns correct error", generate_swagger_example: true do
```

Comment:
In reality `generate_swagger_example: true` is used only once per response type so even when you use it for multiple tests, it will pick just one example and save if for docs

Reply:
ahh, upps